### PR TITLE
Update core-command to use common bootstrap logic

### LINKED ISSUE #1801: Convert core-command to use common bootstrap package
See #1739 for implementation and example conversion of core-data.

Add @michaelestrin and @tsconn23 as reviewers to your pull request.

Address any questions on implementation to @michaelestrin.

### PR DESCRIPTION
**Pull request by AnthonyMBonafide**
Fix #1801

Update main.go and init.go to leverage the common bootstrapping
functionality available to all services and remove redundant code

Update core-command's DBClient to include a 'CloseSession' method to be
used during shut-down.

Signed-off-by: Anthony M. Bonafide <AnthonyMBonafide@gmail.com>

### CODE REVIEW

File: cmd/core-command/main.go [hunk 1]
```diff
@@ -46,49 +44,19 @@ func main() {
...
-		signal.Notify(c, os.Interrupt)
-		errChan <- fmt.Errorf("%s", <-c)
-	}()
+	httpServer := handlers.NewServerBootstrap(command.LoadRestRoutes())
+	bootstrap.Run(
+		configDir,
+		profileDir,
+		internal.ConfigFileName,
+		useRegistry,
+		clients.CoreDataServiceKey,
```
Comment (michaelestrin) [inline cmd/core-command/main.go]: This should be service key for core-command.

File: internal/core/command/config.go [hunk 2]
```diff
@@ -27,3 +30,44 @@ type ConfigurationStruct struct {
 type WritableInfo struct {
 	LogLevel string
 }
+
+func (c *ConfigurationStruct) UpdateFromRaw(rawConfig interface{}) bool {
```
Comment (michaelestrin) [inline internal/core/command/config.go]: Add docblock comments to each of these methods -- prefer you duplicate the comments from other existing implementations (i.e. core-data) for consistency's sake.

File: internal/core/command/init.go [hunk 3]
```diff
@@ -15,218 +15,103 @@
...
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db/redis"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/endpoint"
 )
 
-var Configuration *ConfigurationStruct
+var Configuration = &ConfigurationStruct{}
 var LoggingClient logger.LoggingClient
 var mdc metadata.DeviceClient
 var dbClient interfaces.DBClient
 var registryClient registry.Client
```
Comment (michaelestrin) [inline internal/core/command/init.go]: Eliminate global.  

It's never initialized.  It's used in initializeClients().  Method is passed `registry` which is initialized and can be used in place of the global reference.

File: internal/core/command/init.go [hunk 4]
```diff
@@ -15,218 +15,103 @@
...
-		if configuration.Service.Port == 0 {
-			return nil, errors.New("error reading configuration from Registry")
+	// initialize database.
+	for startupTimer.HasNotElapsed() {
+		var err error
+		dbClient, err = s.newDBClient(Configuration.Databases["Primary"].Type)
+		if err == nil {
+			break
 		}
+		LoggingClient.Warn(fmt.Sprintf("couldn't create database client: %v", err.Error()))
```
Comment (michaelestrin) [inline internal/core/command/init.go]: You need a `dbClient = nil` statement before the logging statement.  See https://github.com/edgexfoundry/edgex-go/issues/1827

File: internal/core/command/init.go [hunk 5]
```diff
@@ -15,218 +15,103 @@
...
+			time.Sleep(time.Second)
 		}
+		LoggingClient.Info("Database disconnected")
+	}()
+
+	return true
+}
+
+// NewServiceInit creates a new ServiceInit
+func NewServiceInit(server server) ServiceInit {
```
Comment (michaelestrin) [inline internal/core/command/init.go]: Would like to see this moved up above Handler for consistency.

File: internal/core/command/init.go [hunk 6]
```diff
@@ -15,218 +15,103 @@
...
+	return ServiceInit{
+		server: server,
 	}
 }
 
-func initializeClients(useRegistry bool) {
+// initializeClients initializes the clients needed by the service.
+func (s ServiceInit) initializeClients(useRegistry bool, registry registry.Client) {
```
Comment (michaelestrin) [inline internal/core/command/init.go]: I wonder if this implementation is correct.

```
func (s ServiceInit) initializeClients(useRegistry bool, registry registry.Client) {
	// Create metadata clients
	params := types.EndpointParams{
		ServiceKey:  clients.CoreMetaDataServiceKey,
		Path:        clients.ApiDeviceRoute,
		UseRegistry: useRegistry,
		Url:         Configuration.Clients["Metadata"].Url() + clients.ApiDeviceRoute,
		Interval:    Configuration.Service.ClientMonitor,
	}

	mdc = metadata.NewDeviceClient(params, endpoint.Endpoint{RegistryClient: &registryClient})
	params.Path = clients.ApiCommandRoute
	params.Url = Configuration.Clients["Metadata"].Url() + clients.ApiCommandRoute
}
```

Maybe auto-merge caused this?  Last two lines/assignments seem unnecessary.

File: internal/core/command/init.go [hunk 7]
```diff
@@ -241,17 +126,11 @@ func initializeClients(useRegistry bool) {
...
-func setLoggingTarget() string {
-	if Configuration.Logging.EnableRemote {
-		return Configuration.Clients["Logging"].Url() + clients.ApiLoggingRoute
-	}
-	return Configuration.Logging.File
-}
-
-func connectToDatabase() error {
+// connectToDatabase creates a DBClient needed to access the underlying database.
+func (s ServiceInit) connectToDatabase() error {
```
Comment (michaelestrin) [inline internal/core/command/init.go]: Unused.  Remove.

**Comment by AnthonyMBonafide** [comment]
Recheck

**Comment by AnthonyMBonafide** [comment]
> Are you able to run the blackbox tests successfully? When I run the core-command-importer collection from edgexfoundry/blackbox-testing's master against edgexfoundry/edgex-go's master (and this) branch, I get a series of 404s.

I noticed the same and thought I was not running something correctly. The cause seems to be that the tests are expecting some HTTP server to accept requests for the `nebulizer` device. I used postman to manually verify the functionality works. I am wondering if something changed in the black-box test setup recently or if I am doing something wrong.

**Comment by michaelestrin** [comment]
@AnthonyMBonafide Please generate issues in https://github.com/edgexfoundry/blackbox-testing and cross-reference here for the problem(s) you had running the blackbox tests.

**Pull request merged**